modules/quality/test/test_brisque.cpp

```diff
@@ -21,13 +21,33 @@ const cv::Scalar
 static const char* MODEL_FNAME = "brisque_model_live.yml";
 static const char* RANGE_FNAME = "brisque_range_live.yml";
 
+// calls cvtest::findDataFile, catches exceptions that we don't want raised and throws a skip test exception
+//  cvtest::findDataFile sometimes throws even if setting required=false
+inline cv::String findDataFile(const cv::String& relative_path, bool required = true)
+{
+    try
+    {
+        return cvtest::findDataFile(relative_path, required);
+    }
+    catch (const SkipTestException&)
+    {
+        throw;//rethrow
+    }
+    catch (...)
+    {
+        if (required)
+            throw;
+        throw SkipTestException("File not found: " + relative_path);
```

Comment:
> cvtest::findDataFile sometimes throws even if setting required=false

It throws on CI (via `OPENCV_TEST_REQUIRE_DATA=1`).
To ensure that all test data files are property committed / deployed (CI should be properly - there were some incidents of using of missing files in tests).

Reply:
That makes sense.  I'll need to add the BRISQUE model data to `opencv_extra` to get the tests to pass, is that OK with you?  Otherwise I'm good skipping the BRISQUE tests if the files aren't there, like we were doing before, and this is my hack to make it happen.  Which do you prefer?

Reply:
Alternatively, if there's a way to reference the OpenCV build directory/samples when the tests are run, we can keep additional data out of `opencv_extra`.  I'll poke around a bit to see if this can happen, but if you have any pointers for me that would be great.  Thanks

Reply:
What is size of this model/models?
We prefer don't commit large blobs into opencv_extra (>10Mb). If these models may be useful for samples, then `opencv_extra` is not a right place too (we prefer to store test data only in opencv_extra).

Additional test data directory can be referenced separately like [DNN tests does](https://github.com/opencv/opencv/blob/3.4.6/modules/dnn/test/test_main.cpp#L11) (files are deployed on CI by maintainers, but we need a link).

For large files (<100Mb) or which should be versioned/updated, we have [opencv_3rdparty](https://github.com/opencv/opencv_3rdparty) repository (we using separate branch for "case" branched from "root" branch). I will provide extra instruction if this is that case.

Reply:
@alalek I have two files, total of 566KB.  

The models are useful not only for the samples, but also for anyone that would want to use the algorithm in production.  This is the more likely scenario vs a user training their own model.  So it seems that `opencv_extra` is probably not the correct place.

`opencv_3rdparty` seems to be the right place if the models must be deployed outside of the `opencv_contrib` build infrastructure

Previously, I had added:

    ocv_add_testdata(samples/ contrib/quality FILES_MATCHING PATTERN "*.yml" )

to my CMakeLists.txt.  This allowed me to search for the model in `{$OPENCV_DIR}/testdata/contrib/quality/` but that only seemed to work on Linux and not Windows, IIRC.  (ie, the `/testdata/` directory isn't being created on Windows during a build/install, unless I'm doing it wrong)

Can you please provide instructions for contributing to the `opencv_3rdparty` repo, if this is still the best path forward?  Thank you

Reply:
Currently check is forced on Linux builders only.
Windows builders just print "skip" message:
```
[ RUN      ] CV_Quality_BRISQUE.static_
[     SKIP ] OpenCV tests: Can't find data file: brisque_range_live.yml
[       OK ] CV_Quality_BRISQUE.static_ (4 ms)
```

OK, files are already committed (opencv_contrib is fine too).

> `ocv_add_testdata(samples/ contrib/quality FILES_MATCHING PATTERN "*.yml" )`

Good!

> brisque_range_live.yml

Should be used as: `contrib/quality/brisque_range_live.yml`

Reply:
or add [searching subdirectory](https://github.com/opencv/opencv_contrib/blob/3.4.6/modules/text/test/test_main.cpp#L8).

Reply:
Thanks @alalek, this seems to be working well for both Win and Linux, at least on my machine.  Hopefully the buildbot agrees when it comes back online